### ISSUE TITLE
No profile visibility settings on Rotten Tomatoes

### ISSUE THREAD

**Issue by funkyaeroplane** [Moglum/rotten-tomatoes-letterboxd#8]
Hi. I'm getting a 401 unauthorised error and I think it's because I can't find anywhere on Rotten Tomatoes where I can edit my profile visibility. Have they removed that setting from the privacy settings page? If so anyway around this?

**Comment by southisup** [comment]
Sadly Fandango seem to be quite aggressively restricting what used to be termed ‘Community’ activity. I discovered in September that I can no longer login to my Rotten Tomatoes account, although I can see it still exists, and can get no response from their helpdesk. Online searches turned up a discussion on Reddit where other people described the same recent experience.

**Comment by funkyaeroplane** [comment]
@southisup yeah it's weird how bad it got. Hence my desire to migrate to Letterboxd. I think Rotten Tomatoes and Flixster were bought and sold a few times and Fandango clearly doesn't care about their users (probably just interested in how they can leverage the tomatometer scores for other things).

I emailed them a while back to ask for a copy of my data in csv and they just said "we don't do that". But it's such a basic service that so many sites offer their users now (especially other rating sites like RYM or Letterboxd). This is one of the many reasons I wanna get away from them but they don't even make that easy :-)

Yeah the login thing is weird now - I can still login but it doesn't use a password anymore - it just emails you a link and after you click it you're logged in (it's worded strangely too as if you are signing up for a new account).

**Comment by southisup** [comment]
I asked for a CSV of my data a few years ago and that was refused too. I just tried logging in again with both my current email and the one I used creating my original Flixster account, and both result in a prompt to create a new account, i.e. names and a new password are requested, so not the same as you're experiencing. I also noticed that although my profile still exists, the reviews page no longer infinite-scrolls, so there's no way I could now access my data beyond the 20 that initially display.

Luckily this tool here worked well enough for me, despite directors export already being broken then (issue #2), that in combination with referring to my intermittent past backups and regularly logging into Rotten Tomatoes, I managed to salvage most of my reviews and ratings. It was a very long, slow slog though.

I made two minor mistakes worth mentioning. The first was not sorting my final spreadsheet into review date order, which resulted in my Letterboxd diary view being in alphabetic order, and rewatch reviews of the same movie being out of sequence on the overview page for the movie – ultimately these are trivial problems because the data is there, correct, if you drill down, just initially confusing to anyone at first glance – which is only likely to ever be myself.

The second mistake was a bit more annoying. I wasn't patient enough reviewing the import process, and confirmed without checking each movie had matched correctly. I've since discovered that their algorithm weights matching title and release year in an unexpected manner (I'd only looked up and added directors for the small minority of movies I anticipated matching wrongly without), and I've later found several where my year differed from theirs by one or two years and my movie ended matched with either something incredibly weird and obscure of the same name, or something completely different that was also released under differing titles around the world, one of which did match mine.

Good luck.

**Comment by funkyaeroplane** [comment]
thanks for the tips.

might not be able to use this solution but I did manage to scroll down all the way to load all my ratings and have it saved as a .html file so maybe I could write some code to search the HTML text: loop through all the ratings and copy data to a csv file. Each movie looks like this:

![image](https://user-images.githubusercontent.com/33277604/208175499-f92b5bee-af10-4ca7-93e3-f91e11e6e1ff.png)

**Comment by southisup** [comment]
Your snippet doesn't have a half-star (star-display__half) in the rating, nor include a review (ratings__comment), and it will be worth comparing some counts to check if yours include any movies in the small number that don't have a release year. But it looks like you have a plan. I used regex extensively during my own cleanup and validation, because that's really the only hammer I'm even half-good with, but I wouldn't relish using that here. Again, good luck.

**Comment by funkyaeroplane** [comment]
@southisup - here's what I ended up doing to recover my ratings. Got most of them - worked well but I seemed to have a few ratings here and there that I know I'd done that weren't in my saved web page of my ratings - not sure if an issue of Rotten Tomatoes not saving my ratings sometimes or if my web page of loaded ratings just didn't load them all or something.

https://github.com/funkyaeroplane/rotten-tomatoes-ratings-to-letterboxd

**Comment by dpi0** [comment]
@funkyaeroplane I am not able load past the first 10 ratings (in the Movie & TV Ratings section) so saving index.html doesn't work

**Comment by dpi0** [comment]
Also https://github.com/sephlietz/rotten-tomatoes-letterboxd/tree/workaround-no-public-option doesn't work either.

As, saving this https://www.rottentomatoes.com/napi/userProfile/movieRatings/USER_ID as input.json works but only for 20 ratings, even after refreshing or waiting it still shows those same 20 ratings. How can i get more ratings?

**Comment by dpi0** [comment]
Has there been a fix for this 401 unauthorized problem?

**Comment by funkyaeroplane** [comment]
> @funkyaeroplane I am not able load past the first 10 ratings (in the Movie & TV Ratings section) so saving index.html doesn't work

Damn! Yeah I had saved my html page of my loaded rating a few months ago (I was noticing the site becoming worse in terms of accessing ratings and logging in, etc. so I thought it could be a good idea). Perhaps it is too late for you for that approach. The thing that bothers me about it all is - even if you want to manually add in all your ratings to Letterboxd, you can't even see them anymore.

I did an online chat and later an email to fandango about the profile visibility thing. Didn't get back to my email. Might hassle them again. If I ever get a response I'll let you know.

**Comment by dpi0** [comment]
> > @funkyaeroplane I am not able load past the first 10 ratings (in the Movie & TV Ratings section) so saving index.html doesn't work
> 
> Damn! Yeah I had saved my html page of my loaded rating a few months ago (I was noticing the site becoming worse in terms of accessing ratings and logging in, etc. so I thought it could be a good idea). Perhaps it is too late for you for that approach. The thing that bothers me about it all is - even if you want to manually add in all your ratings to Letterboxd, you can't even see them anymore.
> 
> I did an online chat and later an email to fandango about the profile visibility thing. Didn't get back to my email. Might hassle them again. If I ever get a response I'll let you know.

Not being able to see more than 10 ratings is terrible not just for backup purposes, but also as a bad user exp in general.
I'll try to email them as well, let's see if they respond or not.

But seeing some reddit threads reporting that their RT accounts are getting deleted for no reasons whatsoever really concerns me as I have around 775 ratings saved in here and now even manual noting & saving the ratings is not an option 😔

**Comment by Olshansk** [comment]
@dvynsh24 I'm in the same boat as you given that I have north of 500 ratings and would like to get access to everything I own. I think I may have found a workound to leverage this library.

1. Go to `https://www.rottentomatoes.com/napi/userProfile/movieRatings/<USER_ID>` replace `<USER_ID>` with your own; [example](https://www.rottentomatoes.com/napi/userProfile/movieRatings/d43956b1-50b8-4729-9728-fdc082537985).

2. Copy as curl from the browser (see the screenshot):

<img width="1493" alt="Screenshot 2023-01-21 at 3 36 41 PM" src="https://user-images.githubusercontent.com/1892194/213886227-05096d0a-2778-40e1-9b15-a4e13e8fe651.png">

3. Replace the request in `index.js` with the following:

```javascript
async function getRatings(cursor) {
  try {
    let res = await rp({
      uri: URI + USER_ID,
      json: true,
      qs: {
        endCursor: cursor,
      },
      headers: {
        cookie: COOKIE,
      },
    });
```

The `COOKIE` value is what I got from the copied curl command

```javascript
const COOKIE =
  'akacd_RTReplatform=2177452799~rv=6~id=999f4c8d7a987446efd176e6b2413ab8; aam_uuid=65435578572463991610595705723651136237; _admrla=2.0-379c1b35-68a5-1108-27cc-37f5a83abbbb; ...
```

3. Running `npm start` managed to generate `rawRatings.json`, but now I'm battling the importer...

**Comment by Olshansk** [comment]
Another couple small hacks inside `createImprotFile`

```
    var date = new Date(age);
    var isoDate =
      date.getFullYear() +
      "-" +
      (date.getMonth() + 1).toString().padStart(2, "0") +
      "-" +
      date.getDate().toString().padStart(2, "0");

    score = score || "";

    comment = comment || "";
```

**Comment by Moglum** [comment]
@Olshansk thank you for taking the time to contribute.
Due to 👨‍👩‍👧‍👦 reasons I don't have any time or energy to spend on this little toy, but I hope it could still help a few more people leave Rotten Tomatoes behind and move to greener pastures.

I have invited you as a Collaborator to this repo if you ever want to contribute more or merge some MRs.

And I have merged your changes in https://github.com/Moglum/rotten-tomatoes-letterboxd/pull/12
